packages/editor/src/utils/with-history/index.js

```diff
@@ -112,6 +112,12 @@ const withHistory = ( options = {} ) => ( reducer ) => {
 			return state;
 		}
 
+		// Temporary: illustrates the problem with RECEIVE_BLOCKS creating an
+		// extra undo level at a random time.
+		if ( action.type === 'RECEIVE_BLOCKS' ) {
```

Comment:
@aduth This fixes the "Should undo to expected level intervals" e2e test for me. The problem seems to be that `RECEIVE_BLOCKS` creates an extra undo level at a random time, when typing happens fast on first page load. I'm not quite sure what this action does in the first place. If this is the point at which we receive content from the server, it seems bad to allow content modification before this action occurs.

Reply:
> Unlike resetBlocks, these should be appended to the existing known set, not replacing.

Huh. What did the user do to cause this?

@noisysocks The only use of this action seems to be reusable blocks. Any idea why its being dispatched when the editor loads, and why it would create an undo level for it?

Reply:
Never mind, resolved it in #11399. :)